src/core/api/link_raw_api.cpp

```diff
@@ -504,9 +509,11 @@ void LinkRaw::HandleEnergyScanTask(void)
             }
         }
 
-        // Post another instance of tha task, since we are
+#if OPENTHREAD_POSIX_VIRTUAL_TIME == 0
```

Comment:
Does this still work if both `OPENTHREAD_CONFIG_ENABLE_SOFTARE_ENERGY_SCAN` and `OPENTHREAD_POSIX_VIRTUAL_TIME` are both 1?

Reply:
Currently travis tests are run with both are 1.

Reply:
It will sample rssi only once in this case, but not multi samples during the duration. This aims to avoid the infinite task post when virtual time is enabled.

Reply:
The energy scan implementation in `mac.cpp` was changed in #2505 to a timer-based one to avoid the infinite task post problem. Should we do the same here?